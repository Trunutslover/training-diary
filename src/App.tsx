import React from 'react';
import './App.module.scss';
import classes from './App.module.scss';
import cn from 'classnames/bind';
import Trainings from "./Components/Trainings/Trainings";
import Header from "./Components/Header/Header";
import {Route, Switch} from "react-router-dom";
import TrainingContainer from "./Components/Training/TrainingContainer";
import Profile from "./Components/Profile/Profile";

const cx = cn.bind(classes);

const App: React.FC = () => {

    return (
        <div className={cx({App: true})}>
            <Header/>
            <Switch>
                <Route path={`/trainings/:trainingId`}>
                    <TrainingContainer />
                </Route>
                <Route path={`/trainings`}>
                    <Trainings />
                </Route>
                <Route path={`/profile`}>
                    <Profile />
                </Route>
                <Route path={`/`}>
                    <div>
                        Hi, this is simple training diary for the bodybuilders. Enjoy!;)
                    </div>
                </Route>
            </Switch>
        </div>
    );
};

export default App;
