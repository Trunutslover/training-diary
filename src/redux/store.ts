import {applyMiddleware, combineReducers, compose, createStore} from "redux";
import trainingsReducer from "./trainingsReducer";
import thunkMiddleware from 'redux-thunk';
import trainingReducer from "./trainingReducer";
import profileReducer from "./profileReducer";
import authReducer from "./authReducer";

const reducers = combineReducers({
    trainings: trainingsReducer,
    training: trainingReducer,
    profile: profileReducer,
    auth: authReducer
});

// @ts-ignore
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const store = createStore(reducers, composeEnhancers(
    applyMiddleware(thunkMiddleware)
));

export default store;