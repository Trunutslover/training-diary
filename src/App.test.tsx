import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import {Provider} from "react-redux";
import store from "./redux/store";
import Enzyme, {shallow, mount} from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import {shallowToJson, mountToJson} from 'enzyme-to-json';

Enzyme.configure({adapter: new Adapter()});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Provider store={store}><App /></Provider>, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe(`App snapshot`, () => {
  it(`App should render correctly`, () => {
    const app = mount(<Provider store={store}><App/></Provider>);
    expect(mountToJson(app)).toMatchSnapshot();
  });
});