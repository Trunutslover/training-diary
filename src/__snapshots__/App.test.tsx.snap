// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot App should render correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <App>
    <div
      className="App"
    >
      Training diary
      <Connect(Trainings)>
        <Trainings
          toggleIsDoneAC={[Function]}
          trainings={
            Array [
              Object {
                "exercises": Array [
                  Object {
                    "name": "squats",
                    "number": 1,
                    "sets": Array [
                      10,
                      12,
                      12,
                      10,
                      8,
                    ],
                  },
                ],
                "id": 1,
                "isDone": false,
              },
            ]
          }
        >
          <div>
            <Training
              exercises={
                Array [
                  Object {
                    "name": "squats",
                    "number": 1,
                    "sets": Array [
                      10,
                      12,
                      12,
                      10,
                      8,
                    ],
                  },
                ]
              }
              id={1}
              isDone={false}
              key="1"
              toggleIsDoneAC={[Function]}
            >
              <section>
                <p>
                  Training number 
                  1
                </p>
                <p>
                  Training is done: 
                  no
                </p>
                <p>
                  Exercises: 
                  <span
                    key="1"
                  >
                    number - 
                    1
                     name - 
                    squats
                     sets - 
                    <span
                      key="0"
                    >
                      10
                       
                    </span>
                    <span
                      key="1"
                    >
                      12
                       
                    </span>
                    <span
                      key="2"
                    >
                      12
                       
                    </span>
                    <span
                      key="3"
                    >
                      10
                       
                    </span>
                    <span
                      key="4"
                    >
                      8
                       
                    </span>
                  </span>
                </p>
                <button
                  onClick={[Function]}
                >
                  Training is done
                </button>
              </section>
            </Training>
          </div>
        </Trainings>
      </Connect(Trainings)>
    </div>
  </App>
</Provider>
`;
